controllers/azurecluster_controller.go

```diff
@@ -97,7 +99,7 @@ func (acr *AzureClusterReconciler) SetupWithManager(ctx context.Context, mgr ctr
 	// Add a watch on clusterv1.Cluster object for unpause notifications.
 	if err = c.Watch(
 		&source.Kind{Type: &clusterv1.Cluster{}},
-		handler.EnqueueRequestsFromMapFunc(util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("AzureCluster"))),
+		handler.EnqueueRequestsFromMapFunc(clusterToInfrastructureMapFunc(ctx, log, mgr.GetClient())),
```

Comment:
wouldn't we need to just add an event filter as in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2023/files#diff-c9f392795aea4d131d38cf4a128c8f6be728ca2ec8b8307cad6ecf1fb7ca5da1L91?

Reply:
The reason a predicate can not be used is that the the annotation only exists on the azurecluster, but not on the capi cluster, so we have to fetch the corresponding azure cluster in order to find out if it has the annotation. That predicate only works for a watch on the azure cluster